vic/drivers/shared_image/src/vic_restore.c

```diff
@@ -0,0 +1,1011 @@
+/******************************************************************************
+ * @section DESCRIPTION
+ *
+ * Read initial model state.
+ *
+ * @section LICENSE
+ *
+ * The Variable Infiltration Capacity (VIC) macroscale hydrological model
+ * Copyright (C) 2014 The Land Surface Hydrology Group, Department of Civil
+ * and Environmental Engineering, University of Washington.
+ *
+ * The VIC model is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *****************************************************************************/
+
+#include <vic_driver_shared_image.h>
+
+/******************************************************************************
+ * @brief    Read initial model state.
+ *****************************************************************************/
+void
+vic_restore(void)
+{
+    extern all_vars_struct    *all_vars;
+    extern domain_struct       global_domain;
+    extern domain_struct       local_domain;
+    extern option_struct       options;
+    extern veg_con_map_struct *veg_con_map;
+    extern filenames_struct    filenames;
+
+    int                        v;
+    size_t                     i;
+    size_t                     j;
+    size_t                     k;
+    size_t                     m;
+    size_t                     p;
+    char                      *cvar = NULL;
+    int                       *ivar = NULL;
+    double                    *dvar = NULL;
+    float                     *fvar = NULL;
+    size_t                     d2count[2];
+    size_t                     d2start[2];
+    size_t                     d3count[3];
+    size_t                     d3start[3];
+    size_t                     d4count[4];
+    size_t                     d4start[4];
+    size_t                     d5count[5];
+    size_t                     d5start[5];
+    size_t                     d6count[6];
+    size_t                     d6start[6];
+
+    // validate state file dimensions and coordinate variables
+    check_init_state_file();
```

Comment:
Exiting from within `check_init_state_file()` is my preference.  The only reason to return a status is if we want to do something other than exit.